docs/project/code-migrate-1.0-sdk.md

```diff
@@ -0,0 +1,159 @@
+
+### Deleted API Operations
+
+The following API operations have been removed:
+
+- `updatePassword`
+- `forgotPassword`
+- `logBranchCommits`
+- `expandTemplate`
+- `createMetaRange`
+- `ingestRange`
+- `updateBranchToken`
+
+### Internal API Operations
+
+The following operations are for `internal` use only and should not be used in your application code. Some deprecated operations have alternatives provided.  
+
+- `setupCommPrefs`
+- `getSetupState`
+- `setup`
+- `getAuthCapabilities`
+- `uploadObjectPreflight`
+- `setGarbageCollectionRulesPreflight`
+- `createBranchProtectionRulePreflight`
+- `postStatsEvents`
+- `dumpRefs` (will be replaced with a long-running API later)
+- `restoreRefs` (will be replaced with a long-running API later)
+- `createSymlinkFile` (Deprecated)
+- `getStorageConfig` (Deprecated. Alternative: `getConfig`)
+- `getLakeFSVersion` (Deprecated. Alternative: `getConfig`)
+- `stageObject` (Deprecated. Alternatives: `get/link physical address` or `import`)
+- `internalDeleteBranchProtectionRule` (Deprecated. Temporary backward support. Alternative: `setBranchProtectionRules`)
+- `internalCreateBranchProtectionRule` (Deprecated. Temporary backward support. Alternative: `setBranchProtectionRules`)
+- `internalGetBranchProtectionRule` (Deprecated. Temporary backward support. Alternative: `getBranchProtectionRules`)
+- `internalDeleteGarbageCollectionRules` (Deprecated. Temporary backward support. Alternative: `deleteGCRules`)
+- `internalSetGarbageCollectionRules` (Deprecated. Temporary backward support. Alternative: `setGCRules`)
+- `internalGetGarbageCollectionRules` (Deprecated. Temporary backward support. Alternative: `getGCRules`)
+- `prepareGarbageCollectionCommits`
+- `getGarbageCollectionConfig`
+
+### New/Updated API Operations
+
+Here are the newly added or updated operations:  
+
+- `getConfig` (Retrieve lakeFS version and storage info)
+- `setBranchProtectionRules` (Route updated)
+- `getBranchProtectionRules` (Route updated)
+- `getGCRules` (New route introduced)
+- `setGCRules` (New route introduced)
+- `deleteGCRules` (New route introduced)
+- `importStatus` (Response structure updated: 'ImportStatusResp' to 'ImportStatus')
+- `uploadObject` (Parameters 'if-none-match' and 'storageClass' are now deprecated)
+- `prepareGarbageCollectionCommits` (Request body removed)
+- `getOtfDiffs` & `otfDiff` (Removed from 'otf diff' tag; retained in 'experimental' tag)
+
+## Migrating SDK Code for Java and JVM-based Languages
+
+### Introduction
+
+If you are using the SDK for Java or any other JVM-based language, be aware that the current generated SDK is not stable with respect to minor version upgrades. Transitioning from `io.lakefs:lakefs-client` to `io.lakefs:sdk` will necessitate rewriting your API calls to fit the new design paradigm.
```

Comment:
Can we use "lakeFS client" for the old version and "SDK" for the new?

Reply:
rewrote, use the term client and package